### PR TITLE
Various WebGL Perf-related changes

### PR DESCRIPTION
**Pull request by jeffsaremi**
See the commit descriptions for details
New Benchmark numbers need to be reproduced

### CODE REVIEW

**Comment by hariharans29** [comment]
Can we also update the corresponding perf improvements in the main README (the image corresponding to GPU latency)

File: lib/backends/webgl/ops/conv.ts [hunk 1]
```diff
@@ -288,15 +289,15 @@ export class WebGLConv extends Conv {
...
+    const preferredBatchSize = 8;
     if (sharedDim < preferredBatchSize || sharedDim % preferredBatchSize !== 0) {
       return sharedDim;
     }
     return preferredBatchSize;
   }
   protected calcBlockSize(outputLayout: TextureLayout): [number, number]|undefined {
-    const preferredRowCount = 64;
-    const preferredColCount = 64;
+    const preferredRowCount = 32;
```
Comment (fs-eire) [inline lib/backends/webgl/ops/conv.ts]: how do we determine the best block size? does it depend on the GPU spec

File: lib/backends/webgl/program-manager.ts [hunk 2]
```diff
@@ -97,6 +101,7 @@ export class ProgramManager {
         } else {
           this.doDraw(buildArtifact, runData);
         }
+        gl.flush();
```
Comment (fs-eire) [inline lib/backends/webgl/program-manager.ts]: we can remove the `gl.flush()` call inside `doDraw()`

**Comment by jeffsaremi** [comment]
I will be re-submitting this after fixing a bug that was introduced.

File: lib/backends/webgl/ops/conv.ts (hunk 1 above)
Comment (jeffsaremi) [reply lib/backends/webgl/ops/conv.ts]: Purely experimental.
These are the best combinations I could come up with on my machine.
In the future, we may decide to expose these to the very savvy users

File: lib/backends/webgl/program-manager.ts (hunk 2 above)
Comment (jeffsaremi) [reply lib/backends/webgl/program-manager.ts]: Thank you. that was left by mistake

**Pull request closed without merging**